osu.Game/Graphics/OsuColour.cs

```diff
@@ -195,6 +195,27 @@ public Color4 ForModType(ModType modType)
             }
         }
 
+        /// <summary>
+        /// Retrieves the accent colour representing a <see cref="Room"/>'s current status.
+        /// </summary>
+        public Color4 ForRoomStatus(Room room)
+        {
+            if (DateTimeOffset.Now >= room.EndDate)
```

Comment:
Having this logic in `OsuColour` also weirds me out a bit. Maybe it's fine, but it's not really where I'd expect to find a date check.

Maybe it'll feel better if moved to a property on `Room`?

`if (room.HasEnded)`

Reply:
It's a bit of a weird one because it needs to be queried periodically. But okay I'll do this for now as a pure code quality improvement.